x/vmbridge/keeper/evm.go

```diff
@@ -56,6 +57,52 @@ func (h SendToWasmEventHandler) Handle(ctx sdk.Context, contract common.Address,
 	return h.Keeper.SendToWasm(ctx, caller, wasmAddr, recipient, amount)
 }
 
+// event __OKCCallToWasm(string wasmAddr,uint256 value, string calldata)
+type CallToWasmEventHandler struct {
+	Keeper
+}
+
+func NewCallToWasmEventHandler(k Keeper) *CallToWasmEventHandler {
+	return &CallToWasmEventHandler{k}
+}
+
+// EventID Return the id of the log signature it handles
+func (h CallToWasmEventHandler) EventID() common.Hash {
+	return types.CallToWasmEvent.ID
+}
+
+// Handle Process the log
+func (h CallToWasmEventHandler) Handle(ctx sdk.Context, contract common.Address, data []byte) error {
+	if !tmtypes.HigherThanEarth(ctx.BlockHeight()) {
+		errMsg := fmt.Sprintf("vmbridger not supprt at height %d", ctx.BlockHeight())
+		return sdkerrors.Wrap(sdkerrors.ErrUnknownRequest, errMsg)
+	}
+
+	params := h.wasmKeeper.GetParams(ctx)
+	if !params.VmbridgeEnable {
+		return types.ErrVMBridgeEnable
+	}
+
+	logger := h.Keeper.Logger()
+	unpacked, err := types.CallToWasmEvent.Inputs.Unpack(data)
+	if err != nil {
+		// log and ignore
+		logger.Error("log signature matches but failed to decode", "error", err)
```

Comment:
why it return nil but not err?

Reply:
it may not be expect event , it may be other event so we can't return other contract normal logic